lib/known_bugs.pm

```diff
@@ -59,4 +62,54 @@ sub create_list_of_serial_failures {
     return $serial_failures;
 }
 
+sub create_list_of_journal_failures {
+    my $journal_failures = [];
+
+    # type=soft will force the testmodule result to softfail
+    # type=hard will just emit a soft fail message but the module will do a normal fail
+
+    # To add a known bug simply copy and adapt the following line:
+    # push @$serial_failures, {type => soft/hard, message => 'Errormsg', pattern => quotemeta 'ErrorPattern' }
+
+    return $journal_failures;
+}
+
+sub upload_journal {
+    my ($file) = @_;
+
+    my $failures = create_list_of_journal_failures();
+
+    $file = upload_logs($file);
+
+    my $die = 0;
+    my %regexp_matched;
+    # loop line by line
+    open(my $journal, '<', "ulogs/$file") or die("Could not open uploaded journal: $file");
+    while (my $line = <$journal>) {
+        chomp $line;
+        for my $regexp_table (@{$failures}) {
+            my $regexp  = $regexp_table->{pattern};
+            my $message = $regexp_table->{message};
+            my $type    = $regexp_table->{type};
+
+            # Input parameters validation
+            die "Wrong type defined for journal failure. Only 'soft' or 'hard' allowed. Got: $type" if $type !~ /^soft|hard|fatal$/;
+            die "Message not defined for journal failure for the pattern: '$regexp', type: $type" if !defined $message;
+
+            # If you want to match a simple string please be sure that you create it with quotemeta
+            if (!exists $regexp_matched{$regexp} and $line =~ /$regexp/) {
+                $regexp_matched{$regexp} = 1;
+                my $fail_type = 'softfail';
+                if ($type eq 'hard') {
+                    record_soft_failure $message. "\n\n" . "$line";
+                }
+                elsif ($type eq 'soft') {
+                    force_soft_failure $message. "\n\n" . "$line";
```

Comment:
this is really bad , you overriding potential test failures just because you found some message in journal ?

Reply:
This is what oli's ticket was about.

Reply:
don't see how this helps . if some ticket saying something does not prove anything

Reply:
I agree with @asmorodskyi here, as we have two types, namely hard and soft, but same behavior. This doesn't make any difference. I would be fine if someone has analyzed case and stated that for it, it doesn't make sense to test anything further. But in case of 'hard' failure, we should fail the test and make it visible. If motivation is review efforts, bug carry over helps enough to evaluate case once. In case you vision it differently, at least remove `type` argument as it's useless with current implementation.

Reply:
Alright, so first of all I've missed that `hard` case has record_soft_failure and not record_soft_failure. So, we have a chance of hiding something only if someone intentionally expects it, which is alright. Second thing is that `force_soft_failure` doesn't override status of the job. So we are good to go and improve it further in next iterations.
So well done ;)